bin/cleanVCF.py

```diff
@@ -1,16 +1,59 @@
 #!/usr/bin/env python
-import sys
 import argparse
+import logging
 
-parser = argparse.ArgumentParser("""removes all vcf entries that are not marked pass or .""")
-parser.add_argument('--vcf',type=str,required=True,help="the path to the vcf file")
-args, unknown = parser.parse_known_args()
-
-for line in open(args.vcf):
-    
-    if(line[0] == "#"):
-        print(line.strip())
-    else:
-        content=line.split("\t")
-        if content[6] == "PASS" or content[6] == "." or content[6] == "pass" or content[6] == "Pass":
-            print(line.strip())
+parser = argparse.ArgumentParser(
+    """
+    Removes the following records from VCF:
+        1. FILTER not marked as . or PASS.
+        2  INFO-field missing CSQ field.
+    """
+)
+parser.add_argument("--vcf", type=str, required=True, help="the path to the vcf file")
+args = parser.parse_args()
+
+logging.basicConfig(
+    level=logging.INFO, format="[%(asctime)s][%(levelname)s]: %(message)s"
+)
+
+
+def csq_missing(info_field):
+    """
+    Variants not passing VEP filters are kept in VCF
+    but not annotated with a CSQ field as of VEP 109.
+
+    Prior to VEP 109 they were removed.
+    """
+    return "CSQ=" not in info_field
+
+
+def filter_pass(filter_field):
+    return filter_field.lower() in ["pass", "."]
+
+
+nbr_filtered = 0
+
+with open(args.vcf) as vcf:
+    for idx, line in enumerate(vcf):
+        line = line.strip()
+
+        if line[0] == "#":
+            print(line)
+            continue
+
+        content = line.split("\t")
+        curr_var = " ".join(content[0:2] + content[5:7])
```

Comment:
I would prefer naming these ranges for readability

For instance `content[6]` vs 

```
info_col_i = 6
content[info_col_i]
```

Reply:
Sorry, `filter_col_i` in this case I guess!

Reply:
Fixed!